src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -32,76 +32,188 @@ import org.graphframes.pattern._
 
 
 /**
+ * A representation of a graph using [[DataFrame]]s.
+ *
  * Represents a graph with vertices and edges stored as [[DataFrame]]s.
  * [[vertices]] must contain a column named "id" storing unique vertex IDs.
  * [[edges]] must contain two columns "src" and "dst" storing source vertex IDs and destination
  * vertex IDs of edges, respectively.
  *
- * @param vertices the [[DataFrame]] holding vertex information
- * @param edges the [[DataFrame]] holding edge information
+ * @groupname structure Structure information
+ * @groupname conversions Conversions
+ * @groupname stdlib Standard graph algorithms
+ * @groupname degree Graph topology
+ * @groupname motif Motif finding
  */
-class GraphFrame protected(
-    @transient val vertices: DataFrame,
-    @transient val edges: DataFrame) extends Logging with Serializable {
+class GraphFrame private(
+    @transient private val _vertices: DataFrame,
+    @transient private val _edges: DataFrame) extends Logging with Serializable {
 
   import GraphFrame._
 
-  require(vertices.columns.contains(ID),
-    s"Vertex ID column '$ID' missing from vertex DataFrame, which has columns: "
-      + vertices.columns.mkString(","))
-  require(edges.columns.contains(SRC),
-    s"Source vertex ID column '$SRC' missing from edge DataFrame, which has columns: "
-      + edges.columns.mkString(","))
-  require(edges.columns.contains(DST),
-    s"Destination vertex ID column '$DST' missing from edge DataFrame, which has columns: "
-      + edges.columns.mkString(","))
-
-  private[graphframes] def sqlContext: SQLContext = vertices.sqlContext
-
   /** Default constructor is provided to support serialization */
   protected def this() = this(null, null)
 
+  // ============== Basic structural methods ============
+
+  /**
+   * The dataframe representation of the vertices of the graph.
+   *
+   * It contains a column called [[GraphFrame.ID]] with the id of the vertex,
+   * and various other user-defined attributes with other attributes.
+   *
+   * The order of the columns is available in [[vertexColumns]].
+   *
+   * @group structure
+   */
+  def vertices: DataFrame = {
+    if (_vertices == null) {
+      throw new Exception("You cannot use GraphFrame objects within a Spark closure")
+    }
+    _vertices
+  }
+
+  /**
+   * The dataframe representation of the edges of the graph.
+   *
+   * It contains two columns called [[GraphFrame.SRC]] and [[GraphFrame.DST]] that contain
+   * the ids of the source vertex and the destination vertex of each edge, respectively.
+   * It may also contain various other columns with user-defined attributes for each edge.
+   *
+   * For symmetric graphs, both pairs src -> dst and dst -> src are present with the same
+   * attributes for each pair.
+   *
+   * The order of the columns is available in [[edgeColumns]].
+   *
+   * @group structure
+   */
+  // TODO(tjh) mention what happens with multiple edges.
```

Comment:
I would not mention this yet.  We should synch with Ankur to see if the treatment of duplicate edges affects the optimizations he's working on.